core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -17,72 +17,125 @@
  * contributors to contribute.
  */
 
-require('domain/opportunity/ExplorationOpportunitySummaryObjectFactory.ts');
-require('domain/opportunity/SkillOpportunityObjectFactory.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ContributionOpportunitiesBackendApiService', [
-  '$http', 'ExplorationOpportunitySummaryObjectFactory',
-  'SkillOpportunityObjectFactory', 'UrlInterpolationService',
-  'OPPORTUNITY_TYPE_SKILL', 'OPPORTUNITY_TYPE_TRANSLATION',
-  'OPPORTUNITY_TYPE_VOICEOVER',
-  function($http, ExplorationOpportunitySummaryObjectFactory,
-      SkillOpportunityObjectFactory, UrlInterpolationService,
-      OPPORTUNITY_TYPE_SKILL, OPPORTUNITY_TYPE_TRANSLATION,
-      OPPORTUNITY_TYPE_VOICEOVER) {
-    var urlTemplate = '/opportunitiessummaryhandler/<opportunityType>';
-    var _getOpportunityFromDict = function(opportunityType, opportunityDict) {
-      if (
-        opportunityType === OPPORTUNITY_TYPE_VOICEOVER ||
-        opportunityType === OPPORTUNITY_TYPE_TRANSLATION) {
-        return ExplorationOpportunitySummaryObjectFactory.createFromBackendDict(
-          opportunityDict);
-      } else if (opportunityType === OPPORTUNITY_TYPE_SKILL) {
-        return SkillOpportunityObjectFactory.createFromBackendDict(
-          opportunityDict);
+import { ExplorationOpportunitySummary } from
+  'domain/opportunity/ExplorationOpportunitySummaryObjectFactory';
+import { SkillOpportunity } from
+  'domain/opportunity/SkillOpportunityObjectFactory';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+const constants = require('constants.ts');
+
+type OpportunityType = 'skill' | 'voiceover' | 'translation';
+
+// TODO(#7165): Replace any with exact type.
+type ContributionOpportunityParams = {
```

Comment:
Can we name this `type` keyword to `interface` please?

Reply:
when I change it to interface, TSLint gives me this on my editor
`No overload matches this call.
The last overload gave the following error.
Type 'ContributionOpportunityParams' is not assignable to type 'HttpParams | { [param: string]: string | string[]; }'.
Type 'ContributionOpportunityParams' is not assignable to type '{ [param: string]: string | string[]; }'.
Index signature is missing in type 'ContributionOpportunityParams'.",`

Reply:
I am getting this error from TSLint when I try to change it to interface.
```No overload matches this call.
  The last overload gave the following error.
    Type 'ContributionOpportunityParams' is not assignable to type 'HttpParams | { [param: string]: string | string[]; }'.
      Type 'ContributionOpportunityParams' is not assignable to type '{ [param: string]: string | string[]; }'.
        Index signature is missing in type 'ContributionOpportunityParams'.```

Reply:
@bansalnitish any idea on what can do about this?

Reply:
Could you please take a look on how interface is created in other PR's? (Example: LearnerAnswerInfoObjectFactory)

Reply:
Again, please feel free to let me know if it still does not work.

Reply:
I looked at the code, I don't think in our codebase we are giving types to any params.

Reply:
@bansalnitish, Can you please check whether this comment is resolved?

Reply:
I'm fine with the code -- marking this as resolved.